packages/arb-bridge-peripherals/contracts/tokenbridge/ethereum/EthERC20Bridge.sol

```diff
@@ -145,48 +143,67 @@ contract EthERC20Bridge is L1Buddy {
         );
     }
 
-    function depositAsERC777(
+    function depositToken(
         address erc20,
         address destination,
         uint256 amount,
         uint256 maxGas,
-        uint256 gasPriceBid
-    ) external payable onlyIfConnected {
+        uint256 gasPriceBid,
+        StandardTokenType tokenType
+    ) private onlyIfConnected {
         require(IERC20(erc20).transferFrom(msg.sender, l2Buddy, amount));
         uint8 decimals = ERC20(erc20).decimals();
+        bytes4 selector;
+        if (tokenType == StandardTokenType.ERC20) {
+            selector = ArbTokenBridge.mintERC20FromL1.selector;
+        } else if (tokenType == StandardTokenType.ERC777) {
+            selector = ArbTokenBridge.mintERC777FromL1.selector;
+        }
         // This transfers along any ETH sent for to pay for gas in L2
         sendPairedContractTransaction(
             maxGas,
             gasPriceBid,
-            abi.encodeWithSelector(
-                ArbTokenBridge.mintERC777FromL1.selector,
-                erc20,
-                destination,
-                amount,
-                decimals
-            )
+            abi.encodeWithSelector(selector, erc20, destination, amount, decimals)
         );
     }
 
+    function depositAsERC777(
+        address erc20,
+        address destination,
+        uint256 amount,
+        uint256 maxGas,
+        uint256 gasPriceBid
+    ) external payable onlyIfConnected {
+        depositToken(erc20, destination, amount, maxGas, gasPriceBid, StandardTokenType.ERC777);
+    }
+
     function depositAsERC20(
         address erc20,
         address destination,
         uint256 amount,
         uint256 maxGas,
         uint256 gasPriceBid
     ) external payable onlyIfConnected {
+        depositToken(erc20, destination, amount, maxGas, gasPriceBid, StandardTokenType.ERC20);
+    }
+
+    function depositAsCustomToken(
```

Comment:
Looks like this is currently private. What the plan for calling it?

Reply:
Also we probably want to restrict this to only working if a custom token has been deployed

Reply:
y and y